### PR TITLE
VehicleSubposition refactoring

### PR DESCRIPTION
**Pull request by frutiemax**
Vehicle position when subposition=0 is unique for each track element type, but when it's different, MiniGolf, GoKarts, Chairlift and ReverserCoaster add some extra data.

So the solution is to add a field in the TrackElementDescriptor for the default subposition and a field in the RideTypeDescriptor for extra vehicle data. I also had to add a VehicleInfoSize since the extra data could have variable array size. New ride types will be able to use that field to add some custom vehicle position data.

### CODE REVIEW

**Comment by frutiemax** [comment]
This one is fixed and merged.

**Comment by frutiemax** [comment]
Rebased

**Comment by frutiemax** [comment]
Rebased, can anyone review this?

File: src/openrct2/ride/coaster/meta/ReverserRollerCoaster.h [hunk 1]
```diff
@@ -51,5 +52,38 @@ constexpr const RideTypeDescriptor ReverserRollerCoasterRTD =
...
+    SET_FIELD(Classification, RideClassification::Ride),
+    SET_FIELD(UpdateLeaveEntrance, PeepUpdateRideLeaveEntranceDefault),
+    SET_FIELD(SpecialElementRatingAdjustment, SpecialTrackElementRatingsAjustment_Default),
+    SET_FIELD(GetGuestWaypointLocation, GetGuestWaypointLocationDefault),
+    SET_FIELD(ConstructionWindowContext, RideConstructionWindowContext::Default),
+    SET_FIELD(RideUpdate, nullptr),
+    SET_FIELD(UpdateMeasurementsSpecialElements, RideUpdateMeasurementsSpecialElements_Default),
+    SET_FIELD(MusicTrackOffsetLength, OpenRCT2::RideAudio::RideMusicGetTrackOffsetLength_Default),
+    SET_FIELD(UpdateRideApproachVehicleWaypoints, UpdateRideApproachVehicleWaypointsDefault),
+    SET_FIELD(VehicleInfoSize, {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 868, 868}),
```
Comment (spacek531) [inline src/openrct2/ride/coaster/meta/ReverserRollerCoaster.h:69]: Is there any reason these values would *not* be the same as the length of the rct_vehicle_info_list?

**Pull request closed without merging**